library/src/main/java/com/esafirm/rxdownloader/RxDownloader.java

```diff
@@ -47,44 +48,108 @@ private DownloadManager getDownloadManager() {
         return downloadManager;
     }
 
-    public Observable<String> download(@NonNull String url,
-                                       @NonNull String filename,
-                                       boolean showCompletedNotification) {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             boolean showCompletedNotification) {
         return download(url, filename, DEFAULT_MIME_TYPE, showCompletedNotification);
     }
 
-    public Observable<String> download(@NonNull String url,
-                                       @NonNull String filename,
-                                       @NonNull String mimeType,
-                                       boolean showCompletedNotification) {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             @NonNull String mimeType,
+                                             boolean showCompletedNotification) {
         return download(createRequest(url, filename, null,
                 mimeType, true, showCompletedNotification));
     }
 
-    public Observable<String> download(@NonNull String url,
-                                       @NonNull String filename,
-                                       @NonNull String destinationPath,
-                                       @NonNull String mimeType,
-                                       boolean showCompletedNotification) {
+    public Single<String> download(@NonNull String url,
+                                             @NonNull String filename,
+                                             @NonNull String destinationPath,
+                                             @NonNull String mimeType,
+                                             boolean showCompletedNotification) {
         return download(createRequest(url, filename, destinationPath,
                 mimeType, true, showCompletedNotification));
     }
 
-    public Observable<String> downloadInFilesDir(@NonNull String url,
-                                                 @NonNull String filename,
-                                                 @NonNull String destinationPath,
-                                                 @NonNull String mimeType,
-                                                 boolean showCompletedNotification) {
+    public Single<String> downloadInFilesDir(@NonNull String url,
+                                                       @NonNull String filename,
```

Comment:
Could you remove funky indentation in here 😄

Reply:
That's strange. May have been Android Studio auto-formatting it. Will check